spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -406,6 +413,30 @@ public Offset latestOffset(Offset startOffset, ReadLimit limit) {
     return latestStreamingOffset.equals(startingOffset) ? null : latestStreamingOffset;
   }
 
+  /**
+   * Get the next snapshot skiping over rewrite and delete snapshots.
+   *
+   * @param curSnapshot the current snapshot
+   * @return the next valid snapshot (not a rewrite or delete snapshot), returns null if all
+   *     remaining snapshots should be skipped.
+   */
+  private Snapshot nextValidSnapshot(Snapshot curSnapshot) {
+    Preconditions.checkArgument(
+        curSnapshot != null, "Sanity check, curSnapshot should not be null");
```

Comment:
i would remove `Sanity check` txt from here as Precondition pretty much applies the same thing